src/api/wallet/WalletApi.ts

```diff
@@ -279,10 +280,10 @@ export class WalletApiImpl implements WalletApi {
 
   /* ****************      Private Functions      **************** */
 
-  private async _notifyListeners(): Promise<void> {
-    await Promise.resolve()
+  private async _notifyListeners(options?: { blockHeader: { number: number } }): Promise<void> {
     const walletInfo: WalletInfo = this.getWalletInfo()
-    this._listeners.forEach(listener => listener(walletInfo))
+    const wInfoExtended = { ...walletInfo, blockNumber: options?.blockHeader?.number }
```

Comment:
Also, I don't get the logic cause I'm not sure how is this used, but in case if we wanted to send the blockNumber, why add the blockHeader in the arguments?

Reply:
Actually, I just copied this form @W3stside pending PR https://github.com/gnosis/dex-react/pull/291

As for `blockHeader`, that's because it contains the `blockNumber`?

Reply:
I'm a bit confused. Maybe we can talk it directly

Reply:
@anxolin 

we want to auto update state right? well that can be done via polling (e.g `setInterval(updateState(), 15seconds)` or via a websocket and some catalyst. In this case, we're using a Websocket connection to `infura` and updating the state when the catalyst updates, which in this case is a new `blockHeader` (`blockNumber`) - this tells us, hey the blockchain was updates, i.e a block was mined, so let's update state.

In conjunction with a `useEffect` hook (or `useMemo`) and setting the dependency as a new block header, we can auto update state that we want without having to manually poll.

Reply:
I understand, but I don't see where this logic is done, I don't find it in the PR. Maybe you can show us in our sync. Also, I don't get why this "blockNumber" needs to be part of the wallet information instead on it's own.

Reply:
All updates happen in [`subscribeToBlockchainUpdate`](https://github.com/gnosis/dex-react/blob/41b8995883b72781b6d0c4396e2ba2a987f38978/src/api/wallet/WalletApi.ts#L118)